### PR TITLE
Added HW1 Iterator Class

### PR DESCRIPTION
**Pull request by putchbes**
Please review my HW1 and leave yo feedback pozyazya

### CODE REVIEW

File: home_work/pavel.ryabinin/hw1/Iterator.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+
+class Iterator {
+    constructor(obj) {
+
+        this[_obj] = obj;
```
Comment (maltez) [inline home_work/pavel.ryabinin/hw1/Iterator.js:5]: Why you do not use . notation?

File: home_work/pavel.ryabinin/hw1/Iterator.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        this.a = 1;
+        this.b = 2;
+    }
+
+    [Symbol.iterator]() {
+             let i = -1;
+             const obj = Object.assign({...this[_obj]}, this);
```
Comment (maltez) [inline home_work/pavel.ryabinin/hw1/Iterator.js:12]: Reason?

**Review by maltez — CHANGES_REQUESTED** [review]

**Comment by putchbes** [comment]
Because I used Symbol to hide property from Iterator. If use dot.notation -
unnable to create Symbol field.

сб, 16 февр. 2019 г. в 18:13, maltez <notifications@github.com>:

File: home_work/pavel.ryabinin/hw1/Iterator.js (hunk 1 above)
Comment (maltez) [reply home_work/pavel.ryabinin/hw1/Iterator.js:5]: Because .notation shortly and this._obj - do not avoid for error in eslint

**Review by maltez — APPROVED** [review]

**Pull request merged**